src/keywords/mod.rs

```diff
@@ -24,12 +26,86 @@ lazy_static! {
 }
 
 /// Keyword with weight
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, PartialEq)]
 pub struct Keyword {
     pub keyword: String,
     pub weight: f64,
 }
 
+#[derive(Debug)]
+pub struct KeywordExtractConfig {
+    stop_words: BTreeSet<String>,
+    min_keyword_length: usize,
+    use_hmm: bool,
+}
+
+impl KeywordExtractConfig {
+    /// Creates a KeywordExtractConfig state that contains filter criteria as
+    /// well as segmentation configuration for use by keyword extraction
+    /// implementations.
+    pub fn new(stop_words: BTreeSet<String>, min_keyword_length: usize, use_hmm: bool) -> Self {
```

Comment:
I'd remove getters (not really useful) and use builder pattern for `KeywordExtractConfig`, like

```rust
let config = KeywordExtractConfig::builder()
    .add_stop_word("word")
    .use_hmm(true)
    // and other options
    .build();
```

or without a separate builder type:

```rust
let config = KeywordExtractConfig::default()
    .add_stop_word("word")
    .use_hmm(true)
    // and other options
    ;
```

Reply:
We can change this later, does not need to block merging this PR.

Reply:
Added #101 to track.